opta/core/kubernetes.py

```diff
@@ -66,24 +66,63 @@ def get_required_path_executables(cloud: str) -> FrozenSet[str]:
     return frozenset({"kubectl"}) | exec_map.get(cloud, set())
 
 
-def configure_kubectl(layer: "Layer") -> None:
-    """Configure the kubectl CLI tool for the given layer"""
+def set_kube_config(layer: "Layer") -> None:
+    """Create a kubeconfig file to connect to a kubernetes cluster specified in a given layer"""
     # Make sure the user has the prerequisite CLI tools installed
     # kubectl may not *technically* be required for this opta command to run, but require
     # it anyways since user must install it to access the cluster.
     ensure_installed("kubectl")
     makedirs(GENERATED_KUBE_CONFIG_DIR, exist_ok=True)
     if layer.cloud == "aws":
-        _aws_configure_kubectl(layer)
+        _aws_set_kube_config(layer)
     elif layer.cloud == "google":
-        _gcp_configure_kubectl(layer)
+        _gcp_set_kube_config(layer)
     elif layer.cloud == "azurerm":
-        _azure_configure_kubectl(layer)
+        _azure_set_kube_config(layer)
     elif layer.cloud == "local":
-        _local_configure_kubectl(layer)
+        _local_set_kube_config(layer)
 
 
-def _local_configure_kubectl(layer: "Layer") -> None:
+def purge_opta_kube_config(layer: "Layer") -> None:
+    """Delete the kubeconfig file created for a given layer, and also remove it from the default kubeconfig file"""
+    config_file_name = get_config_file_name(layer)
+    opta_config: dict
+    if exists(config_file_name):
+        opta_config = yaml.load(open(config_file_name))
+        remove(config_file_name)
+    else:
+        return
+
+    default_kube_config_filename = expanduser(
+        KUBE_CONFIG_DEFAULT_LOCATION.split(ENV_KUBECONFIG_PATH_SEPARATOR)[0]
+    )
+    if not exists(default_kube_config_filename):
+        return
+
+    default_kube_config = yaml.load(open(default_kube_config_filename))
+    opta_config_user = opta_config["users"][0]
+    opta_config_context = opta_config["contexts"][0]
+    opta_config_cluster = opta_config["clusters"][0]
+    for opta_value, key in [
+        [opta_config_user, "users"],
+        [opta_config_context, "contexts"],
+        [opta_config_cluster, "clusters"],
+    ]:
+        current_indices = [
+            i
+            for i, x in enumerate(default_kube_config[key])
+            if x["name"] == opta_value["name"]
+        ]
+        for index in sorted(current_indices, reverse=True):
+            del default_kube_config[key][index]
+
+    if default_kube_config["current-context"] == opta_config_context["name"]:
+        default_kube_config["current-context"] = ""
+    with open(default_kube_config_filename, "w") as f:
+        yaml.dump(default_kube_config, f)
```

Comment:
this block could be extracted out in it's own generic method `kubeconfig_unset(filename, user: Optional[str], context: Optional[str], cluster: Optional[str])`
such method could be unit tested without any opta objects